get-started/part2.md

```diff
@@ -136,149 +140,130 @@ def hello():
-```
+Python serves your app at http://0.0.0.0:80, but that message comes from
+_inside_ the container, so Python does not know you remapped the port to 4000.
+Remapping here demonstrates the difference between what you `EXPOSE` within the
+`Dockerfile`, and what you _publish_ using `docker run -p`.
 
-This port remapping of `4000:80` is to demonstrate the difference
-between what you `EXPOSE` within the `Dockerfile`, and what you `publish` using
-`docker run -p`. In later steps, we'll just map port 80 on the host to port 80
-in the container and use `http://localhost`.
+> Expected Redis error
+>
+> Because Redis is not running (as we only installed the Python library, and
+> not Redis itself), the attempt to use it here fails. We fix that in [Part 5](part5.md).
 
-Hit `CTRL+C` in your terminal to quit.
+Press `CTRL+C` in your terminal to quit.
 
- > On Windows, explicitly stop the container
- >
- > On Windows systems, `CTRL+C` does not stop the container. So, first
- type `CTRL+C` to get the prompt back (or open another shell), then type
- `docker container ls` to list the running containers, followed by
- `docker container stop <Container NAME or ID>` to stop the
- container. Otherwise, you'll get an error response from the daemon
- when you try to re-run the container in the next step.
+> On Windows, press `CTRL+C` to get the prompt back, then manually stop the
+> running container:
+> ```shell
+> docker container ls
+> docker container stop <CONTAINER ID or NAMES>
+> ```
 
-Now let's run the app in the background, in detached mode:
+Run the app again in _detached_ mode (with `-d`). The full CONTAINER ID displays
+while the container runs in the background:
 
 ```shell
-docker run -d -p 4000:80 friendlyhello
+$ docker run -d -p 4000:80 hellomoby
+7c96ed69a93403785f49b321447f2754d320aaf76fce2fd41e09da62d92af8bf
 ```
 
-You get the long container ID for your app and then are kicked back to your
-terminal. Your container is running in the background. You can also see the
-abbreviated container ID with `docker container ls` (and both work interchangeably when
-running commands):
+List the abbreviated CONTAINER ID, NAMES, and other info:
 
 ```shell
 $ docker container ls
-CONTAINER ID        IMAGE               COMMAND             CREATED
-1fa4ab2cf395        friendlyhello       "python app.py"     28 seconds ago
+CONTAINER ID      IMAGE          COMMAND              CREATED             NAMES
+7c96ed69a934      hellomoby      "python app.py"      28 seconds ago      vigilant_mccarthy
 ```
 
-You'll see that `CONTAINER ID` matches what's on `http://localhost:4000`.
-
-Now use `docker container stop` to end the process, using the `CONTAINER ID`, like so:
+Refresh the browser at http://localhost:4000 and notice that "Hostname" reflects
+the new `CONTAINER ID`. Stop the container using `CONTAINER ID` or `NAMES`:
 
 ```shell
-docker container stop 1fa4ab2cf395
+$ docker container stop vigilant_mccarthy
```

Comment:
When you created the container, if you named it, you would be able to reliably use the container name in these commands.

Reply:
The container was automatically named ... not sure I understand.

Reply:
If you let it be automatically named, the user following along with you won't ever see exactly the same output as you and might get confused or make a copy-paste error. If you force the container to have a name using `--name` then you and the user following along can always be on the same page.